lib/InfluxRequest.js

```diff
@@ -106,20 +107,23 @@ InfluxRequest.prototype._request = function (options, callback) {
   // need to store the original path, in case we need to re-submit the request onError and change the host
   if (!requestOptions.originalUrl) requestOptions.originalUrl = requestOptions.url
   requestOptions.url = this.url(host, requestOptions.originalUrl)
-  requestOptions.retries++
-  request(requestOptions, function (err, response, body) {
-    self._parseCallback(err, response, body, requestOptions, callback)
-  })
-}
-
-InfluxRequest.prototype._parseCallback = function (err, response, body, requestOptions, callback) {
-  if (err && resubmitErrorCodes.indexOf(err.code) !== -1) {
-    this.disableHost(requestOptions.host)
-    if (this.options.maxRetries >= requestOptions.retries && this.hostIsAvailable()) {
-      return this._request(requestOptions, callback)
+  requestOptions.retries++  // fetch() ignores the `.retries` option
+  var theResponse
+  fetch(requestOptions.url, requestOptions).then(function (response) {
+    theResponse = response
+    return response.text()
+  }).then(function (body) {
+    body = body === '' ? {} : JSON.parse(body)  // workaround for https://github.com/bitinn/node-fetch/issues/165
```

Comment:
I agree it's less resilient, but as far as adding it in node-fetch:

 - it is a breaking API change
 - empty objects are not necessarily equivalent to empty responses
 - it breaks away from the specification which node-fetch is trying to match (!)
 - it also will not be a function in browsers; even if the `fetch` module for browsers also accepts such a change, it won't be present in browsers on which the polyfill is not applied (modern Chrome/FireFox)